app/scripts/ChromeSerialBridge.js

```diff
@@ -74,29 +75,49 @@ export default class ChromeSerialBridge {
       });
 
       serialPort.on('data', function (data) {
-        console.log('serialport data');
         var resp = {};
         resp.op = 'data';
         resp.data = data;
         port.postMessage(resp);
       });
 
-      port.onMessage.addListener(function (msg) {
-        console.log('socket received data');
+      function trySend(buffer) {
+        if (buffer) {
+          queue.push(buffer);
+        }
+
+        if (queue.length === 0) {
+          // Exhausted pending send buffer.
+          return;
+        }
 
-        // console.log('socket received', msg);
+        if (queue.length > 512) {
+          throw new Error('Send queue is full! More than 512 pending messages.');
```

Comment:
Any particular reason for 512?

Reply:
In practice I never saw the retry queue get more than a few entries deep... mainly wanted an escape hatch in case all the packets become undeliverable for some reason!

Probably should extract this magic number as a constant...